docs/contributor/image_mode.md

```diff
@@ -0,0 +1,308 @@
+this configuration is inherited by the container so that it can be used by the
+MicroShift CSI driver to allocate storage.
+
+```
+$ sudo vgs
+  VG   #PV #LV #SN Attr   VSize   VFree
+  rhel   1   1   0 wz--n- <91.02g <2.02g
+```
+
+Additional configuration is required for using storage in `bootc` MicroShift
+images when the host does not have the volume group setup as required by the
+MicroShift CSI driver.
+
+Run the following commands to create a file to be used for LVM partitioning and
+set it up as a loop device.
+
+```
+VGFILE=/var/lib/microshift-lvm-storage.img
+VGSIZE=1G
+
+sudo truncate --size="${VGSIZE}" "${VGFILE}"
+sudo losetup -f "${VGFILE}"
+```
+
+Query the loop device name and create a free volume group on the device according
+to the MicroShift CSI driver requirements.
+
+```
+VGLOOP=$(losetup -j ${VGFILE} | cut -d: -f1)
+sudo vgcreate -f -y rhel "${VGLOOP}"
+```
+
+The device will now be shared with the newly created containers as described in
+the next section.
+
+> The following commands can be run to detach the loop device and delete the LVM
+> volume group file.
+> ```
+> sudo losetup -d "${VGLOOP}"
+> sudo rm -f "${VGFILE}"
+> ```
+
+### Run Container
+
+Run the following command to start the MicroShift `bootc` image in an interactive
+terminal session. Note that the host container storage and kernel modules paths
+are shared with the container.
+
+```
+IMAGE_NAME=microshift-4.16-bootc
+
+sudo podman run --rm -it --privileged \
+    -v /var/lib/containers/storage:/var/lib/containers/storage \
+    -v /lib/modules:/lib/modules:ro \
+    --name "${IMAGE_NAME}" \
+    "${IMAGE_NAME}"
+```
+
+> If additional LVM volume group device was allocated as described in the
+> [Configure CSI](#configure-csi) section, the loop device should automatically
+> be shared with the container and used by the MicroShift CSI driver.
+
+Log into the container host when prompted using the `redhat:redhat` credentials.
+Run the following command to verify that all the MicroShift pods are up and running
+without errors.
+
+```
+sudo oc get pods -A -w \
+    --kubeconfig /var/lib/microshift/resources/kubeadmin/kubeconfig
+```
+
+> Run the `sudo shutdown now` command to stop the container host.
+
+## Run MicroShift Bootc Virtual Machine
+
+Log into the `physical hypervisor host` using the user credentials that have
+SUDO permissions configured.
+
+### Prepare Kickstart File
+
+Create the `kickstart.ks` file with the following contents to be used during the
+virtual machine installation.
+
+```
+lang en_US.UTF-8
+keyboard us
+timezone UTC
+text
+reboot
+
+# Partition disk with a 1GB boot XFS partition and an LVM volume containing a 10GB+
+# system root. The remainder of the volume is for the CSI driver for storing data.
+zerombr
+clearpart --all --initlabel
+part biosboot --fstype=biosboot --size=1 --asprimary
+part /boot/efi --fstype=efi --size=200
+part /boot --fstype=xfs --asprimary --size=800
```

Comment:
I would recommend just: `reqpart --add-boot`. xref https://github.com/osbuild/images/pull/363/commits/79662c40ea25b4e456e05b4684ec2c0bab6f322a for example

Reply:
Great suggestion! If this works for MicroShift, I would replace it "everywhere". 
Created [USHIFT-3789](https://issues.redhat.com/browse/USHIFT-3789) issue to follow-up on this.